src/main/java/com/stripe/net/APIResource.java

```diff
@@ -132,6 +134,24 @@ protected static String instanceURL(Class<?> clazz, String id, String apiBase)
 		}
 	}
 
+	protected static String subresourceURL(Class<?> clazz, String id, Class<?> subClazz)
+			throws InvalidRequestException {
+		return subresourceURL(clazz, id, subClazz, Stripe.getApiBase());
+	}
+
+
+	protected static String subresourceURL(Class<?> clazz, String id, Class<?> subClazz, String apiBase)
+			throws InvalidRequestException {
+		try {
+			return String.format("%s/%s/%ss", classURL(clazz, apiBase), urlEncode(id), className(subClazz));
+		} catch (UnsupportedEncodingException e) {
+			throw new InvalidRequestException("Unable to encode parameters to "
+					+ CHARSET
+					+ ". Please contact support@stripe.com for assistance.",
+					null, null, null, 0, e);
+		}
+	}
```

Comment:
Hm, I'm not too crazy about adding a couple new global helpers like this and then using them in just the one place. It feels like we should either just re-use the template found elsewhere (see transfer reversals for example), or convert everything over to use these.

What do you think @ob-stripe?

Reply:
And actually, looking at what's already in here, it seems like we just have subresources in transfer reversals, external accounts, and application fee refunds, so might be plausible to just move everything.

Reply:
Yeah, I feel the same. I'm fine with adding a new helper method but let's try and make its use consistent.

> transfer reversals, external accounts, and application fee refunds

Also customer sources.

Reply:
So TL;DR; is to move those to the `subresourceURL` helper? If yes I think it's reasonable to do this in another PR. wdyt?